feathr_project/feathr/client.py

```diff
@@ -348,27 +359,27 @@ def get_offline_features(self,
         feature_queries = feature_query if isinstance(feature_query, List) else [feature_query]
         config = tm.render(feature_lists=feature_queries, observation_settings=observation_settings, output_path=output_path)
         config_file_name = "feature_join_conf/feature_join.conf"
-        config_file_path = os.path.abspath(config_file_name)
+        config_file_path = os.path.join(self.local_workspace_dir, config_file_name)
         write_to_file(content=config, full_file_name=config_file_path)
-        return self._get_offline_features_with_config(config_file_name)
+        return self._get_offline_features_with_config(config_file_path)
 
     def _get_offline_features_with_config(self, feature_join_conf_path='feature_join_conf/feature_join.conf'):
         """Joins the features to your offline observation dataset based on the join config.
 
         Args:
           feature_join_conf_path: Relative path to your feature join config file.
         """
-        if not feature_join_conf_path.startswith('feature_join_conf'):
-            raise RuntimeError(
-                'Feature join config should be in feature_join_conf folder.')
 
-        _FeatureRegistry.save_to_feature_config(Path("./"))
+        if 'anchor_list' in dir(self) and 'derived_feature_list' in dir(self):
+            _FeatureRegistry.save_to_feature_config_from_context(self.anchor_list, self.derived_feature_list, self.local_workspace_dir)
+        else:
+            RuntimeError("Please call FeathrClient.build_features() first in order to materialize the features")
```

Comment:
This API is not to materialize features. Fix the error message?

Reply:
Fixec